### PR TITLE
Middleware unit tests and test context for integration tests

### PR DESCRIPTION
**Pull request by djlmills**

### CODE REVIEW

File: packages/interbit-ui-tools/src/tests/middleware/middleware.test.js [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    const store = instrumentedStore()
+
+    store.dispatch(
+      actionCreators.chainLoaded({
+        chainAlias,
+        chainId
+      })
+    )
+
+    store.setupDone()
```
[the remark this answers is not in the document]
Comment (djlmills) [reply packages/interbit-ui-tools/src/tests/middleware/middleware.test.js]: There are a few tests that require specific preconditions, such as having the chains loaded or blocking that cause actions to be dispatched or state changes. `store.setupDone()` resets the store, so only new actions, chain actions and state changes are tracked.

File: packages/interbit-test/test/interbit.test.js [hunk 2]
```diff
@@ -50,16 +50,12 @@ describe('interbit', () => {
...
-    // regular function is required for before to honour timeout
-    // eslint-disable-next-line prefer-arrow-callback
-    before(async function() {
-      // Timeout extended for 2048 bit key generation
-      this.timeout(20000)
+    beforeAll(async () => {
       keyPair = await interbit.generateKeyPair()
       hypervisor = await interbit.createHypervisor({ keyPair })
-    })
+    }, 20000) // Timeout extended for 2048 bit key generation
```
Comment (nicholeous) [inline packages/interbit-test/test/interbit.test.js]: Please move magic numbers up into consts

File: packages/interbit-test/test/interbit.test.js [hunk 3]
```diff
@@ -150,28 +146,32 @@ describe('interbit', () => {
...
+                unsubscribe()
+                console.timeEnd('Time to unsubscribe')
+              }
+            })
+            // Potentially brittle
+            // Test needs to wait for at least 2 blocks
+            // sleep timeout assumes a blocking frequency of 2 secs
+            // test timeout needs to be longer than the sleep period
+            // Added more wiggle room around timeouts for Heroku
+            await sleep(5000)
```
Comment (nicholeous) [inline packages/interbit-test/test/interbit.test.js]: const

File: packages/interbit-test/test/interbit.test.js [hunk 4]
```diff
@@ -150,28 +146,32 @@ describe('interbit', () => {
...
+            assert.equal(count, 1)
+          },
+          8000
```
Comment (nicholeous) [inline packages/interbit-test/test/interbit.test.js]: const

**Pull request merged**